crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -706,14 +706,20 @@ impl CallChainLayout {
     }
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, is_macro::Is)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub(crate) enum OwnParentheses {
     /// The node has parentheses, but they are empty (e.g., `[]` or `f()`).
     Empty,
     /// The node has parentheses, and they are non-empty (e.g., `[1]` or `f(1)`).
     NonEmpty,
 }
 
+impl OwnParentheses {
+    const fn is_non_empty(self) -> bool {
+        matches!(self, OwnParentheses::NonEmpty)
```

Comment:
nit: one magic macro less

```suggestion
        self == OwnParentheses::NonEmpty
```

Reply:
`PartialEq` is not const :(

Reply:
The compiler knows that these are identical: https://godbolt.org/z/Monn4Y53T

Reply:
That's correct but it means that any method using `is_non_empty` can't be `const` either...

Reply:
do we need it to be const, given that the function is transparent to the compiler either way?

Reply:
I prefer keeping functions const if we can because it allows using const on call sites which may enable compile time optimisations that the compiler otherwise would not do (because the function isn't marked as `const`).